functions/src/index.ts

```diff
@@ -48,35 +48,31 @@ export const getHintKit = onFlow(
   }
 );
 
-export const resetGame = onDocumentUpdated(
-  "boardInfo/solvedWordsCount",
+export const resetGame = onDocumentCreated(
+  "solvedWords/{docId}",
   async (event) => {
     try {
-      const data = event.data?.after.data();
-      const previousData = event.data?.before.data();
-
-      if (data?.value == previousData?.value) {
-        return null;
-      }
-
       const db = getFirestore();
 
-      const collectionRef = db.collection("boardInfo");
+      const solvedWordsCollection = db.collection("solvedWords");
+      const solvedWordsSnapshot = await solvedWordsCollection.get();
+      const solvedWordsCount = solvedWordsSnapshot.size;
```

Comment:
The best way to implement this would be a `count` query, here's the [documentation](https://firebase.google.com/docs/firestore/query-data/aggregation-queries#use_the_count_aggregation) for it.
so we can do:
`const solvedWordsCount = await solvedWordsCollection.count().get();`

Reply:
Nice! Do you mind updating it? 🙏🏼